addons/mrp/migrations/15.0.2.1/pre-migration.py

```diff
@@ -0,0 +1,8 @@
+def migrate(cr, version):
+    cr.execute(
+        """
+        UPDATE mrp_production as mp
+           SET date_start = (SELECT MIN(date_start) FROM mrp_workorder WHERE production_id = mp.id)
+         WHERE mp.state != 'draft' AND mp.date_start IS NULL
```

Comment:
If we finally allow for a migration script.

This query would run a subquery for each updated mrp_production row. For huge DBs it could be problematic. I would suggest using a CTE.
/!\Untested suggestion below/!\
```suggestion
        WITH info AS (
            SELECT mp.id,
                   MIN(wo.date_start) AS date_start
              FROM mrp_production mp
              JOIN mrp_workorder wo
                ON wo.production_id = mp.id
             WHERE mp.state != 'draft'
               AND mp.date_start IS NULL
             GROUP BY mp.id
        ) UPDATE mrp_production mp
             SET date_start = info.date_start
            FROM info
           WHERE mp.id = info.id
```

Reply:
@aj-fuentes Thanks for the suggestion :+1: